components/brave_wallet_ui/page/container.tsx

```diff
@@ -444,123 +444,47 @@ function Container (props: Props) {
-                  selectedBTCAssetPrice={selectedBTCAssetPrice}
-                  selectedAssetPriceHistory={formatedPriceHistory}
-                  portfolioPriceHistory={portfolioPriceHistory}
-                  selectedTimeline={selectedTimeline}
-                  selectedPortfolioTimeline={selectedPortfolioTimeline}
-                  transactions={transactions}
-                  userAssetList={userAssetList}
-                  fullAssetList={fullTokenList}
-                  onConnectHardwareWallet={onConnectHardwareWallet}
-                  onAddHardwareAccounts={onAddHardwareAccounts}
-                  getBalance={getBalance}
-                  onCreateAccount={onCreateAccount}
-                  onImportAccount={onImportAccount}
-                  isLoading={isFetchingPriceHistory}
-                  showAddModal={showAddModal}
-                  onToggleAddModal={onToggleAddModal}
-                  onUpdateAccountName={onUpdateAccountName}
-                  onUpdateVisibleTokens={onUpdateVisibleTokens}
-                  fetchFullTokenList={fetchFullTokenList}
-                  userWatchList={userVisibleTokens}
-                  selectedNetwork={selectedNetwork}
-                  onSelectNetwork={onSelectNetwork}
-                  isFetchingPortfolioPriceHistory={isFetchingPortfolioPriceHistory}
-                  onRemoveAccount={onRemoveAccount}
-                  privateKey={privateKey ?? ''}
-                  onDoneViewingPrivateKey={onDoneViewingPrivateKey}
-                  onViewPrivateKey={onViewPrivateKey}
-                  onImportAccountFromJson={onImportAccountFromJson}
-                  onSetImportError={onSetImportError}
-                  hasImportError={importError}
-                  onAddCustomToken={onAddCustomToken}
-                  transactionSpotPrices={transactionSpotPrices}
-                  userVisibleTokensInfo={userVisibleTokensInfo}
-                />
-              )}
-            </>
-          )}
-        </>
-      )
+    // If wallet is not yet created will Route to Onboarding
+    if ((!isWalletCreated || setupStillInProgress) && walletLocation !== WalletRoutes.Restore) {
+      history.push(WalletRoutes.Onboarding)
+      // If wallet is created will Route to login
+    } else if (isWalletLocked && walletLocation !== WalletRoutes.Restore) {
+      history.push(WalletRoutes.Unlock)
+      // Allowed Private Routes if a wallet is unlocked else will redirect back to Portfolio
+    } else if (
+      !isWalletLocked &&
+      hasInitialized &&
+      walletLocation !== WalletRoutes.Backup &&
+      walletLocation !== WalletRoutes.Accounts &&
+      acceptedAccountRoutes.length !== 0 &&
+      !acceptedAccountRoutes.includes(walletLocation) &&
+      walletLocation !== WalletRoutes.Portfolio &&
+      acceptedPortfolioRoutes.length !== 0 &&
+      !acceptedPortfolioRoutes.includes(walletLocation)
+    ) {
+      history.push(WalletRoutes.Portfolio)
```

Comment:
I feel this is hard to understand, and we might forget to update this block as newer routes are introduced. Is there a way to rewrite this in a better way?

I'm thinking about it too, but if we don't find a good way now, we can deal with it later on.